ShareContacts/explainer.md

```diff
@@ -0,0 +1,91 @@
+
+## Goals
+
+* Allow installed web applications to participate in contacts/people UX across different platforms.
+
+## Non-goals
+
+*
+
+## Use Cases
+
+### Integrating with platform's Share UX
+
+A user decides to share content and opens the Share UX on the platform. They are presented with their frequent contacts on several apps. They can click on a contact to share the content with that contact. Overall, this represents a faster an easier way to share content. This is also behaviour that is present in several desktop and mobile platforms already. It saves the steps of having to open the app and look for the contact you will be sharing with. The following image shows how the share UX shows contacts on Windows and Android. Notice that each contact is associated to a specific application, as represented by a small app-icon on the bottom-right corner of the displayed contact.
+
+![Share UX on multiple platforms](contactShare.png)
+
+## Proposed solution
+
+The proposed solution is a promise-based method that works in scope of an installed web application to share a list of contacts with the underlying platform. This gets the contacts-per-app into the OS. The other part of the feature is akin to Web Share Target, except it is targeted directly at a specific contact.
+
+### Providing Contacts to the platform
+
+`navigator.shareContacts(<contacts>);`
+
+The method receives contacts as a list of JSON objects. Each object must have at least an `id`, `firstName` and `displayPicture`. Optionally, the contact can have a `lastName`, `rank` and a list of `phones` and `emails`.
+
+#### Contact detailed fields
+
+* `id`: unique id for the contact.
+* `firstName`: Contact's first name(s).
+* `displayPicture`: url to the display picture of the contact.
+* `lastName`: Contact's last name(s).
+* `rank`: integer that defines the rank or precedence that a contact has in the list. This could be used by the platform to display top contacts first.
+* `phones`: list of strings that represent phone numbers.
+* `emails`: list of strings that represent email addresses.
+
+```json
+{
+    "id" : "contact_id_1",
+    "firstName" : "John",
+    "lastName" : "Doe",
+    "displayPicture" : "https://app.com/imgs/usrs/contactPic.jpg",
+    "rank" : 1,
+    "phones" : ["(+44)7123456789"],
+    "emails" : ["j.doe@home.com", "j.doe@work.com"]
+}
+
+```
+
+### Registering the Contact as a share target for the installed app
+
+TODO: investigate how this works on the native side
```

Comment:
Based on the Windows and Android versions, it seems like the appropriate approach would be a new ShareTargetParams member. Potentially `contacts` as an array of contact IDs?

https://w3c.github.io/web-share-target/#sharetargetparams-and-its-members